reference/service/webhooks.md

```diff
@@ -0,0 +1,254 @@
+
+| Event Kind | Triggered |
+| `stack` | Whenever a stack is created or deleted within an organization. |
+| `team`  | Whenever a team is created, updated, or deleted. |
+
+## Payloads
+
+Each webhook payload has a format specific to the payload being emitted. Every payload will contain a sender, organization, and stack reference.
+
+Each webhook should contain a `user` field, which is the user who requested the action, an `organization` which is
+the organization name and URL for the event. And finally the `stackName` for the stack which was modified as applicable.
+
+### Headers
+
+Payloads also contain several headers.
+
+| Header | Description |
+| `Pulumi-Webhook-ID` | Unique ID for each webhook sent. For reference when looking at delivery logs in the Pulumi Cloud Console. |
+| `Pulumi-Webhook-Kind` | The kind of webhook event, e.g. `stack_update`. |
+| `Pulumi-Webhook-Signature` | Only set if the webhook has a shared secret. HMAC hex digest of the request payload, using the `sha256` hash function and the webhook secret as the HMAC key. |
+
+<p></p>
+
+The following snippets show how to compute and verify the webhook signature.
+For examples in other languages, see [danharper/hmac-examples](https://github.com/danharper/hmac-examples).
+
+{% include langchoose.html %}
+
+```javascript
+var crypto = require('crypto');
+
+const sharedSecret = ...
+const payload = req.body.toString();
+
+var hmacAlg = crypto.createHmac('sha256', sharedSecret)
+var expectedSignature = hmac.update(payloadBody).digest('hex');
+```
+
+```typescript
+import * as crypto from "crypto";
+
+const sharedSecret = ...
+const payload = req.body.toString();
+
+const hmacAlg = crypto.createHmac("sha256", stackConfig.sharedSecret);
+const hmac = hmacAlg.update(payload).digest("hex");
+
+const result = crypto.timingSafeEqual(Buffer.from(webhookSig), Buffer.from(hmac));
+```
+
+```python
+import hashlib
+import hmac
+import base64
+
+message = bytes('...', 'utf-8')
+secret = bytes('...', 'utf-8')
+
+hash = hmac.new(secret, message, hashlib.sha256)
+hash.hexdigest()
+
+expected_signature = base64.b64encode(hash.digest())
+```
+
+```go
+func computeSignature(payload []byte, secret string) string {
+	mac := hmac.New(sha256.New, []byte(secret))
+	_, err := mac.Write(payload)
+	contract.AssertNoErrorf(err, "computing HMAC digest")
+	return fmt.Sprintf("%x", mac.Sum(nil))
+}
+```
+
+## Payload Reference
+
+Most payloads contain `user` and `organization` fields. `user` contains the
+identity of the user who triggered the webhook, e.g. the person who initiated
+the stack update or performed the action.
+
+> The Pulumi Webhook payloads are under development, and may be changed from
+> time to time. See the [Pulumi Community Slack](http://slack.pulumi.io/) for
+> any announcements or changes.
+
+### `stack` event
+
+```
+{
+	"user": {
+		"name": "Morty Smith",
+		"githubLogin": "morty",
+		"avatarUrl": "https://crazy-adventures.net/morty.png",
+	},
+	"organization": "cazy-adventures",
```

Comment:
So incredibly minor, but: `"crazy-adventures"`. (There are four instances of `cazy`, looks like.)

Reply:
🤦‍♂️ done.